### ISSUE TITLE
Support parameter splitting with commas for applicable search parameters

### ISSUE THREAD

**Issue by BrokenEagle** [danbooru/danbooru#4227]
For an existing usecase where this would be useful, there is the userscript ValidateTagInput (see [topic #14474](https://danbooru.donmai.us/forum_topics/14474)). It searches **tag aliases** by `antecedent_name` and **tag implication**s by `consequent_name` when validating tag adds and removes. This can currently require a dozen up to over a hundred individual network calls. If this could be done instead with a single network call, that would save on both client and server resources.

Those are currently the only two examples I use myself, but I imagine that this could be useful elsewhere, such as wherever search parameters can be queried against the database using a list of values. An example would be searching instances by names (name, title, creator name, updater name, approver name, antecedent name, consequent name) when they have such,  and ID fields whenever numeric IDs are involved. There's probably more examples that I'm not thinking of right now, but that's a good place to start from.

**Comment by evazion** [comment]
This is already possible for numeric fields. For text fields it depends on whether commas are valid characters. Commas are valid characters in tag aliases, so spaces are used instead.

https://danbooru.donmai.us/tag_aliases?search[antecedent_name]=neko+cats
https://danbooru.donmai.us/tag_aliases?search[consequent_name]=cat+dog
https://danbooru.donmai.us/tag_implications?search[consequent_name]=cat+dog
https://danbooru.donmai.us/tag_aliases?search[name_matches]=*,*

**Comment by BrokenEagle** [comment]
Just Discovered a useful search that doesn't support spaces or commas, which would be searching by multiple titles on the `wiki_pages` endpoint.

I have an immediate usecase where this would prove useful, namely adding subscribe links on the wiki pages index for the *EventListener* userscript. Beforehand I scraped the wiki page ID from the index table which was easy since the links still had the ID in the URL. Since that was changed with 3a908f84bbf5169659ca2b2b96e088a1d17ccf78, that is no longer an option. Instead, I'll have to resort to doing up to 20 queries per page load instead of being able to do it in one.
